core/templates/packages.el.template

```diff
@@ -1,31 +1,63 @@
-;;; packages.el --- %LAYERNAME% Layer packages File for Spacemacs
+;;; packages.el --- %LAYER_NAME% layer packages file for Spacemacs.  -*- lexical-binding: t; -*-
```

Comment:
In a lot of case it's not needed, and I'm afraid it would make it difficult to allow user to customize behaviours later (explained [here](http://www.emacswiki.org/emacs/DynamicBindingVsLexicalBinding#toc4)). But I'm not an expert so I may be wrong. @TheBB, @syl20bnr?